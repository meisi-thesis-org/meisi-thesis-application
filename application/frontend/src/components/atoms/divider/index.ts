import DividerComponent from './divider.component.vue';

export {
  DividerComponent
}
