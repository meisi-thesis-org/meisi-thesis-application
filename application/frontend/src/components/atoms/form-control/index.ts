import type { FormControlComponentProps } from './form-control.component.type';
import FormControlComponent from './form-control.component.vue';

export {
  type FormControlComponentProps,
  FormControlComponent
}
