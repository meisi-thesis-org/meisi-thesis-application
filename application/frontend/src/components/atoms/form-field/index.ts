import type { FormField } from './form-field.component.type';

export type {
  FormField
}
