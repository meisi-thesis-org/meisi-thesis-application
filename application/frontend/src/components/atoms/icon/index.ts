import IconComponent from './icon.component.vue';

export {
  IconComponent
}
