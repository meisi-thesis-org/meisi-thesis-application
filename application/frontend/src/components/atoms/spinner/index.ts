import SpinnerComponent from './spinner.component.vue';

export { SpinnerComponent };
