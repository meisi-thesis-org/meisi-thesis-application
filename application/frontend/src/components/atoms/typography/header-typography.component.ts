import { type HTMLTemplateResult, LitElement, html, css, type CSSResultGroup } from 'lit';
import { customElement, property } from 'lit/decorators.js';

@customElement('app-header-typography')
export class HeaderTypographyComponent extends LitElement {
  @property({ type: String })
  public content: string = '';

  public static override styles?: CSSResultGroup | undefined = css`
    .typography {
      font-size: clamp(1rem, 1rem + 0.5vw, 2rem);
      font-weight: 800;
    }
  `;

  public override render(): HTMLTemplateResult {
    return html`
      <div class="typography">
        ${this.content}
      </div>
    `;
  }
}
