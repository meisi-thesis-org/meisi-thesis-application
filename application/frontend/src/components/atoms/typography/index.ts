import TypographyComponent from './typography.component.vue';

export {
  TypographyComponent
}
