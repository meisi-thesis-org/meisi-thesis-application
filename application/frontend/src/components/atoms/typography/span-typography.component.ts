import { type HTMLTemplateResult, LitElement, html, css, type CSSResultGroup } from 'lit';
import { customElement, property } from 'lit/decorators.js';

@customElement('app-span-typography')
export class SpanTypographyComponent extends LitElement {
  @property({ type: String })
  public content: string = '';

  public static override styles?: CSSResultGroup | undefined = css`
    .typography {
      font-size: clamp(0.70rem, 0.70rem + 0.5vw, 1.70rem);
      font-weight: 800;

    }
  `;

  public override render(): HTMLTemplateResult {
    return html`
      <div class="typography">
        ${this.content}
      </div>
    `;
  }
}
