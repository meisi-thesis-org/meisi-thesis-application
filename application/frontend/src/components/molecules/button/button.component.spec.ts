import { afterEach, describe, expect, it, vi } from 'vitest';
import { type VueWrapper, shallowMount } from '@vue/test-utils';
import { ButtonComponent } from '.';
import { TypographyComponent } from './../../atoms/typography';

describe('ButtonComponent', () => {
  afterEach(() => {
    vi.clearAllMocks()
  })

  function callShallowMount (): VueWrapper {
    return shallowMount(ButtonComponent, {
      props: {
        content: 'dummyContent',
        action: () => {}
      }
    })
  }

  it('should match the snapshot', () => {
    expect(callShallowMount()).toMatchSnapshot();
  })

  describe('typography', () => {
    it('should exist', () => {
      const wrapper = callShallowMount();
      expect(wrapper.findComponent(TypographyComponent).exists()).toBe(true);
    })

    it('should have content attribute', () => {
      const wrapper = callShallowMount();
      expect(wrapper.findComponent(TypographyComponent).attributes().content).toBe('dummyContent');
    })
  })
})
