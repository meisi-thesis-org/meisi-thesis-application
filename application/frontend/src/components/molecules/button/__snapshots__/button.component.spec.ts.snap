// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonComponent > should match the snapshot 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="button"
        data-v-7a12e89c=""
      >
        <typography.component-stub
          content="dummyContent"
          data-v-7a12e89c=""
          segment="sub-header"
        />
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "slotsProxy": null,
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "attrsProxy": null,
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "action": [Function],
            "content": "dummyContent",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "TypographyComponent": {
              "__file": "/Users/diogo.oliveira/Documents/@development/@meisi-thesis/application/frontend/src/components/atoms/typography/typography.component.vue",
              "__name": "typography.component",
              "__scopeId": "data-v-395649b3",
              "props": {
                "content": {
                  "required": true,
                  "type": [Function],
                },
                "segment": {
                  "required": true,
                  "type": [Function],
                },
              },
              "render": [Function],
              "setup": [Function],
            },
            "definedProps": {
              "action": [Function],
              "content": "dummyContent",
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "action": [Function],
            "content": "dummyContent",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "action": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "content": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
            },
            [],
          ],
          "provides": {},
          "proxy": {
            "action": [Function],
            "content": "dummyContent",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "TypographyComponent": {
              "__file": "/Users/diogo.oliveira/Documents/@development/@meisi-thesis/application/frontend/src/components/atoms/typography/typography.component.vue",
              "__name": "typography.component",
              "__scopeId": "data-v-395649b3",
              "props": {
                "content": {
                  "required": true,
                  "type": [Function],
                },
                "segment": {
                  "required": true,
                  "type": [Function],
                },
              },
              "render": [Function],
              "setup": [Function],
            },
            "definedProps": {
              "action": [Function],
              "content": "dummyContent",
            },
          },
          "slots": {},
          "slotsProxy": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": {
                  "a": null,
                  "accessCache": {},
                  "appContext": {
                    "app": [Circular],
                    "components": {
                      "transition": {
                        "name": "transition",
                      },
                      "transition-group": {
                        "name": "transition-group",
                      },
                    },
                    "config": {
                      "compilerOptions": {},
                      "errorHandler": undefined,
                      "globalProperties": {},
                      "isNativeTag": [Function],
                      "optionMergeStrategies": {},
                      "performance": false,
                      "warnHandler": undefined,
                    },
                    "directives": {},
                    "emitsCache": WeakMap {},
                    "mixins": [
                      {
                        "beforeCreate": [Function],
                      },
                    ],
                    "optionsCache": WeakMap {},
                    "propsCache": WeakMap {},
                    "provides": {},
                    "reload": [Function],
                  },
                  "asyncDep": null,
                  "asyncResolved": false,
                  "attrs": {},
                  "attrsProxy": null,
                  "bc": null,
                  "bm": null,
                  "bu": null,
                  "bum": null,
                  "c": null,
                  "components": null,
                  "ctx": {
                    "content": "dummyContent",
                    "segment": "sub-header",
                  },
                  "da": null,
                  "data": {},
                  "directives": null,
                  "ec": null,
                  "effect": ReactiveEffect {
                    "active": true,
                    "allowRecurse": true,
                    "deps": [
                      Set {
                        [Circular],
                      },
                      Set {
                        [Circular],
                      },
                      Set {
                        [Circular],
                      },
                    ],
                    "fn": [Function],
                    "onTrack": undefined,
                    "onTrigger": undefined,
                    "parent": undefined,
                    "scheduler": [Function],
                  },
                  "emit": [Function],
                  "emitsOptions": null,
                  "emitted": null,
                  "exposeProxy": null,
                  "exposed": null,
                  "inheritAttrs": undefined,
                  "isDeactivated": false,
                  "isMounted": true,
                  "isUnmounted": false,
                  "m": null,
                  "next": null,
                  "parent": [Circular],
                  "props": {
                    "content": "dummyContent",
                    "segment": "sub-header",
                  },
                  "propsDefaults": {},
                  "propsOptions": [
                    {
                      "content": {
                        "0": false,
                        "1": true,
                        "required": true,
                        "type": [Function],
                      },
                      "segment": {
                        "0": false,
                        "1": true,
                        "required": true,
                        "type": [Function],
                      },
                    },
                    [],
                  ],
                  "provides": {},
                  "proxy": {
                    "content": "dummyContent",
                    "segment": "sub-header",
                  },
                  "refs": {},
                  "render": [Function],
                  "renderCache": [],
                  "root": [Circular],
                  "rtc": null,
                  "rtg": null,
                  "scope": EffectScope {
                    "_active": true,
                    "cleanups": [],
                    "detached": true,
                    "effects": [
                      ReactiveEffect {
                        "active": true,
                        "allowRecurse": true,
                        "deps": [
                          Set {
                            [Circular],
                          },
                          Set {
                            [Circular],
                          },
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "onTrack": undefined,
                        "onTrigger": undefined,
                        "parent": undefined,
                        "scheduler": [Function],
                      },
                    ],
                    "parent": undefined,
                  },
                  "setupContext": {
                    "attrs": {},
                    "emit": [Function],
                    "expose": [Function],
                    "slots": {},
                  },
                  "setupState": {},
                  "slots": {},
                  "slotsProxy": {},
                  "sp": null,
                  "subTree": {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <typography.component-stub
                      content="dummyContent"
                      data-v-7a12e89c=""
                      segment="sub-header"
                    />,
                    "key": null,
                    "patchFlag": 0,
                    "props": {
                      "content": "dummyContent",
                      "segment": "sub-header",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "typography.component-stub",
                  },
                  "suspense": null,
                  "suspenseId": 0,
                  "type": {
                    "model": undefined,
                    "name": "typography.component",
                    "props": {
                      "content": {
                        "required": true,
                        "type": [Function],
                      },
                      "segment": {
                        "required": true,
                        "type": [Function],
                      },
                    },
                    "setup": [Function],
                  },
                  "u": null,
                  "uid": 2,
                  "um": null,
                  "update": [Function],
                  "vnode": [Circular],
                  "withProxy": null,
                },
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": [
                  "content",
                ],
                "el": <typography.component-stub
                  content="dummyContent"
                  data-v-7a12e89c=""
                  segment="sub-header"
                />,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "content": "dummyContent",
                  "segment": "sub-header",
                },
                "ref": null,
                "scopeId": "data-v-7a12e89c",
                "shapeFlag": 4,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": {
                  "model": undefined,
                  "name": "typography.component",
                  "props": {
                    "content": {
                      "required": true,
                      "type": [Function],
                    },
                    "segment": {
                      "required": true,
                      "type": [Function],
                    },
                  },
                  "setup": [Function],
                },
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": {
                  "a": null,
                  "accessCache": {
                    "@@__IMMUTABLE_ITERABLE__@@": 0,
                    "@@__IMMUTABLE_RECORD__@@": 0,
                    "_isMockFunction": 0,
                    "constructor": 0,
                    "content": 3,
                    "segment": 3,
                    "toJSON": 0,
                    Symbol(Symbol.toStringTag): 0,
                  },
                  "appContext": {
                    "app": [Circular],
                    "components": {
                      "transition": {
                        "name": "transition",
                      },
                      "transition-group": {
                        "name": "transition-group",
                      },
                    },
                    "config": {
                      "compilerOptions": {},
                      "errorHandler": undefined,
                      "globalProperties": {},
                      "isNativeTag": [Function],
                      "optionMergeStrategies": {},
                      "performance": false,
                      "warnHandler": undefined,
                    },
                    "directives": {},
                    "emitsCache": WeakMap {},
                    "mixins": [
                      {
                        "beforeCreate": [Function],
                      },
                    ],
                    "optionsCache": WeakMap {},
                    "propsCache": WeakMap {},
                    "provides": {},
                    "reload": [Function],
                  },
                  "asyncDep": null,
                  "asyncResolved": false,
                  "attrs": {},
                  "attrsProxy": {},
                  "bc": null,
                  "bm": null,
                  "bu": null,
                  "bum": null,
                  "c": null,
                  "components": null,
                  "ctx": {
                    "content": "dummyContent",
                    "segment": "sub-header",
                  },
                  "da": null,
                  "data": {},
                  "directives": null,
                  "ec": null,
                  "effect": ReactiveEffect {
                    "active": true,
                    "allowRecurse": true,
                    "deps": [
                      Set {
                        [Circular],
                      },
                      Set {
                        [Circular],
                      },
                      Set {
                        [Circular],
                      },
                    ],
                    "fn": [Function],
                    "onTrack": undefined,
                    "onTrigger": undefined,
                    "parent": undefined,
                    "scheduler": [Function],
                  },
                  "emit": [Function],
                  "emitsOptions": null,
                  "emitted": null,
                  "exposeProxy": null,
                  "exposed": null,
                  "inheritAttrs": undefined,
                  "isDeactivated": false,
                  "isMounted": true,
                  "isUnmounted": false,
                  "m": null,
                  "next": null,
                  "parent": [Circular],
                  "props": {
                    "content": "dummyContent",
                    "segment": "sub-header",
                  },
                  "propsDefaults": {},
                  "propsOptions": [
                    {
                      "content": {
                        "0": false,
                        "1": true,
                        "required": true,
                        "type": [Function],
                      },
                      "segment": {
                        "0": false,
                        "1": true,
                        "required": true,
                        "type": [Function],
                      },
                    },
                    [],
                  ],
                  "provides": {},
                  "proxy": {
                    "content": "dummyContent",
                    "segment": "sub-header",
                  },
                  "refs": {},
                  "render": [Function],
                  "renderCache": [],
                  "root": [Circular],
                  "rtc": null,
                  "rtg": null,
                  "scope": EffectScope {
                    "_active": true,
                    "cleanups": [],
                    "detached": true,
                    "effects": [
                      ReactiveEffect {
                        "active": true,
                        "allowRecurse": true,
                        "deps": [
                          Set {
                            [Circular],
                          },
                          Set {
                            [Circular],
                          },
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "onTrack": undefined,
                        "onTrigger": undefined,
                        "parent": undefined,
                        "scheduler": [Function],
                      },
                    ],
                    "parent": undefined,
                  },
                  "setupContext": {
                    "attrs": {},
                    "emit": [Function],
                    "expose": [Function],
                    "slots": {},
                  },
                  "setupState": {},
                  "slots": {},
                  "slotsProxy": {},
                  "sp": null,
                  "subTree": {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <typography.component-stub
                      content="dummyContent"
                      data-v-7a12e89c=""
                      segment="sub-header"
                    />,
                    "key": null,
                    "patchFlag": 0,
                    "props": {
                      "content": "dummyContent",
                      "segment": "sub-header",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "typography.component-stub",
                  },
                  "suspense": null,
                  "suspenseId": 0,
                  "type": {
                    "model": undefined,
                    "name": "typography.component",
                    "props": {
                      "content": {
                        "required": true,
                        "type": [Function],
                      },
                      "segment": {
                        "required": true,
                        "type": [Function],
                      },
                    },
                    "setup": [Function],
                  },
                  "u": null,
                  "uid": 2,
                  "um": null,
                  "update": [Function],
                  "vnode": [Circular],
                  "withProxy": null,
                },
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": [
                  "content",
                ],
                "el": <typography.component-stub
                  content="dummyContent"
                  data-v-7a12e89c=""
                  segment="sub-header"
                />,
                "key": null,
                "patchFlag": 8,
                "props": {
                  "content": "dummyContent",
                  "segment": "sub-header",
                },
                "ref": null,
                "scopeId": "data-v-7a12e89c",
                "shapeFlag": 4,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": {
                  "model": undefined,
                  "name": "typography.component",
                  "props": {
                    "content": {
                      "required": true,
                      "type": [Function],
                    },
                    "segment": {
                      "required": true,
                      "type": [Function],
                    },
                  },
                  "setup": [Function],
                },
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="button"
              data-v-7a12e89c=""
            >
              <typography.component-stub
                content="dummyContent"
                data-v-7a12e89c=""
                segment="sub-header"
              />
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": {
              "class": "button",
            },
            "ref": null,
            "scopeId": "data-v-7a12e89c",
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/diogo.oliveira/Documents/@development/@meisi-thesis/application/frontend/src/components/molecules/button/button.component.vue",
            "__name": "button.component",
            "__scopeId": "data-v-7a12e89c",
            "components": {},
            "props": {
              "action": {
                "required": true,
                "type": [Function],
              },
              "content": {
                "required": true,
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="button"
          data-v-7a12e89c=""
        >
          <typography.component-stub
            content="dummyContent"
            data-v-7a12e89c=""
            segment="sub-header"
          />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "action": [Function],
          "content": "dummyContent",
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Users/diogo.oliveira/Documents/@development/@meisi-thesis/application/frontend/src/components/molecules/button/button.component.vue",
          "__name": "button.component",
          "__scopeId": "data-v-7a12e89c",
          "components": {},
          "props": {
            "action": {
              "required": true,
              "type": [Function],
            },
            "content": {
              "required": true,
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="button"
          data-v-7a12e89c=""
        >
          <typography.component-stub
            content="dummyContent"
            data-v-7a12e89c=""
            segment="sub-header"
          />
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.4",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <div
    class="button"
    data-v-7a12e89c=""
  >
    <typography.component-stub
      content="dummyContent"
      data-v-7a12e89c=""
      segment="sub-header"
    />
  </div>,
}
`;
